pkg/mcshub/controller/controller_test.go

```diff
@@ -97,411 +82,291 @@ func TestImportGenerated(t *testing.T) {
-	compareSi(&excepted2, &si, assertions)
-} */
-
-// Pass - I still needs to make sure it really testing this feature ok, didn't had the time to make sure its ok
-func Test2ImportsFromExports(t *testing.T) {
-	exp1, es1 := prepareServiceExport(t)
-	exp2, es2 := prepareServiceExport(t)
-	exp2.Name = serviceName2                                      // won't let me add 2 serviceExports with same name - probably not the field I should have edit.
-	exp2.Labels[lhconst.LighthouseLabelSourceName] = serviceName2 // changed field
-	exp2.Labels[lhconst.LighthouseLabelSourceCluster] = cluster2
+// since we're using a fake client and not a real k8s API endpoint, the import is not
+// garbage collected. We do validate the finalizer is cleared...
+func TestCreateAndDeleteExport(t *testing.T) {
+	assert := require.New(t)
+	exp, err := prepareServiceExport(export, cluster1)
+	assert.Nil(err)
 
-	preloadedObjects := []runtime.Object{service, exp1, exp2}
+	preloadedObjects := []runtime.Object{exp}
 	ser := controller.ServiceExportReconciler{
 		Client: getClient(preloadedObjects),
 		Log:    newLogger(t, false),
 		Scheme: getScheme(),
 	}
-
-	req1 := reconcile.Request{
-		NamespacedName: types.NamespacedName{
-			Name:      exp1.GetName(),
-			Namespace: exp1.GetNamespace(),
-		}}
-
-	result, err := ser.Reconcile(context.TODO(), req1)
-
-	if err != nil {
-		t.Error(err)
-	}
-
-	if result.Requeue {
-		t.Error("unexpected requeue")
-	}
-
-	// check si1:
-	si := mcsv1a1.ServiceImport{}
-	err = ser.Client.Get(context.TODO(), req1.NamespacedName, &si)
-
-	if err != nil {
-		t.Error(err)
-	}
-	assertions := require.New(t)
-	excepted := generateExceptedServiceImport(exp1, es1)
-	compareSi(&excepted, &si, assertions)
-
-	// check si2:
-	req2 := reconcile.Request{
+	req := reconcile.Request{
 		NamespacedName: types.NamespacedName{
-			Name:      exp2.GetName(),
-			Namespace: exp2.GetNamespace(),
+			Name:      exp.GetName(),
+			Namespace: exp.GetNamespace(),
 		}}
 
-	result, err = ser.Reconcile(context.TODO(), req2)
+	_, err = ser.Reconcile(context.TODO(), req)
+	assert.Nil(err)
 
-	if err != nil {
-		t.Error(err)
-	}
+	err = ser.Client.Get(context.TODO(), req.NamespacedName, exp)
+	assert.Nil(err)
 
-	if result.Requeue {
-		t.Error("unexpected requeue")
-	}
+	now := metav1.Now()
+	exp.SetDeletionTimestamp(&now) // emulate a delete by setting the timestamp
+	err = ser.Client.Update(context.TODO(), exp, &client.UpdateOptions{})
+	assert.Nil(err)
 
-	err = ser.Client.Get(context.TODO(), req2.NamespacedName, &si)
+	_, err = ser.Reconcile(context.TODO(), req)
+	assert.Nil(err)
 
-	if err != nil {
-		t.Error(err)
-	}
+	exp = &mcsv1a1.ServiceExport{} // start with a fresh copy so we don't see stale data
+	err = ser.Client.Get(context.TODO(), req.NamespacedName, exp)
+	assert.Nil(err)
 
-	excepted2 := generateExceptedServiceImport(exp2, es2)
-	compareSi(&excepted2, &si, assertions)
+	assert.Len(exp.GetFinalizers(), 0, "expected no finalizers, found", exp.GetFinalizers())
```

Comment:
can also be extracted into a func with expected finalizer count parameter

Reply:
it would not be any shorter or clearer, IMO.